modules/nf-core/plink/exclude/main.nf

```diff
@@ -0,0 +1,40 @@
+process PLINK_EXCLUDE {
+    tag "$meta.id"
+    label 'process_low'
+
+    conda "bioconda::plink=1.90b6.21"
+    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
+        'https://depot.galaxyproject.org/singularity/plink:1.90b6.21--h779adbc_1' :
+        'quay.io/biocontainers/plink:1.90b6.21--h779adbc_1' }"
+
+    input:
+    tuple val(meta), path(bed), path(bim), path(fam), path(variants)
+
+    output:
+    tuple val(meta), path("*.bed"), emit: bed
+    tuple val(meta), path("*.bim"), emit: bim
+    tuple val(meta), path("*.fam"), emit: fam
+    path "versions.yml"           , emit: versions
+
+    when:
+    task.ext.when == null || task.ext.when
+
+    script:
+    def args = task.ext.args ?: ''
+    def prefix = task.ext.prefix ?: "${meta.id}"
+    if( "$bed" == "${prefix}.bed" ) error "Input and output names are the same, use \"task.ext.prefix\" to disambiguate!"
+    """
+    plink \\
+        --bfile ${meta.id} \\
+        $args \\
+        --exclude $variants \\
+        --threads $task.cpus \\
+        --make-bed \\
```

Comment:
Is `--make_bed` an argument that always has to be given when running this tool?

Reply:
In fact, no. My first approach when designing this module was to pass it as an `ext.args`, following nf-core guidelines. However, I saw that `plink extract` has this argument explicit and I tried to be consistent between similar modules. As your input here is a bed-bim-fam and you would like to extract/exclude variants from this triset, the `--make-bed` forces the output to keep being a bed-bim-fam. I'd like your opinion on this, but I think that I'll be better to add the `--make-bed` to the `ext.args`.

Reply:
I think you should keep it in `ext.args` and adjust your output channels to also work when this argument isn't given (I assume a BED file isn't created if this argument isn't given?)

Reply:
If you don't provide  `--make-bed` the program breaks with "`Warning: No output requested.  (Did you forget --make-bed?)  Exiting.` " `--make-bed`  makes the bed-bim-fam, not only the bed.

Reply:
Yes, you need to commonly pass  `--make-bed`  to get the output or the command `--recode` to transform the output to another file type. Thanks for your feedback! :)

Reply:
Ah so there is an option to create another file type? Then you should adjust the module to work for both :)